doc/extdev/appapi.rst

```diff
@@ -125,6 +125,26 @@ package.
    .. versionchanged:: 0.5
       Added the support for keyword arguments giving visit functions.
 
+.. method:: Sphinx.add_enumerable_node(node, figtype, title_getter=None, **kwds)
+
+   Register a Docutils node class as a numfig target.  Sphinx numbers the node
+   automatically. And then the users can refer it using :rst:role:`numref`.
+   figure, table or code-block. And then the node is numbered automatically.
+
+   *figtype* should be one of ``figure``, ``table``, ``code-block`` or any
+   other figure types.
```

Comment:
IMO, "figure", "table" and "code-block" figtypes are provided by sphinx as default and each figtypes have individual numbering sequence. Users can assign original nodes to such default figtypes to follow the existent numbering sequence. Otherwise, users can assign original nodes to new figtype by this API and the new figtype will have another numbering sequence. 
It would be better if this story is described in document.